Sources/MapboxDirections/RouteOptions.swift

```diff
@@ -281,20 +281,21 @@ open class RouteOptions: DirectionsOptions {
         if let speed = speed {
             params.append(URLQueryItem(name: CodingKeys.speed.stringValue, value: String(speed)))
         }
-        
-        if !roadClassesToAvoid.isEmpty && roadClassesToAvoid.isDisjoint(with: [.highOccupancyVehicle2, .highOccupancyVehicle3, .highOccupancyToll]) {
-            let allRoadClasses = roadClassesToAvoid.description.components(separatedBy: ",").filter { !$0.isEmpty }
-            precondition(allRoadClasses.count < 2, "You can only avoid one road class at a time.")
-            if let firstRoadClass = allRoadClasses.first {
-                params.append(URLQueryItem(name: CodingKeys.roadClassesToAvoid.stringValue, value: firstRoadClass))
+
+        if !roadClassesToAvoid.isEmpty {
+            if !roadClassesToAvoid.isSubset(of: .validAvoidClasses) {
+                assertionFailure("Provided road classes cannot be avoided")
```

Comment:
In general, we avoid client-side validation, except for options that don’t make logical sense.

Reply:
Agree, removed the assertion.